### DISCUSSION TITLE
ARC-0004: Flagged Operations

### DISCUSSION THREAD

**Discussion by bendyarm** [ProvableHQ/ARCs#63] [ARCs]
This is a discussion section for ARC-0004: Flagged Operations
---
arc: 4
title: Flagged Operations
authors: @bendyarm, @d0cd, @acoglio
discussion: ARC-0004: Flagged 
topic: Language
status: Draft
created: 2024-01-17
---

## Abstract

Some Aleo instructions can halt when passed certain arguments.
This halting behavior makes those instructions unusable
by a program that wants to try something else if a halt is detected.

For example, if a Leo program contains a conditional not in a
finalize, both branches of the conditional are executed in the
circuit, so if a branch not taken executes an instruction that halts,
the whole program will incorrectly halt.  Replacing the halting
instruction by one that returns an error flag will allow such a
program to compile correctly.

This ARC proposes to add new opcodes for flagged operations
corresponding to the Aleo Instructions that can halt.

**Comment by zklim** [comment]
Duplicated https://github.com/ProvableHQ/ARCs/discussions/89
